pkg/occlient/occlient.go

```diff
@@ -1502,27 +1541,38 @@ func (c *Client) UpdateDCToSupervisor(ucp UpdateComponentParams, isToLocal bool,
 		inputEnvs = deleteEnvVars(inputEnvs, EnvS2ISrcBackupDir)
 	}
 
-	// Generate the SupervisorD Config
-	dc := generateSupervisordDeploymentConfig(
-		ucp.CommonObjectMeta,
-		ucp.ImageMeta,
-		inputEnvs,
-		cmpContainer.EnvFrom,
-		&ucp.ResourceLimits,
-	)
+	var dc appsv1.DeploymentConfig
+	if isCreatePVC {
```

Comment:
what happens if `isCreatePVC` isn't true? Would it be correct in saying that supervisord won't be added at all?

Reply:
If I'm not mistaken it is an indicator that we are updating from git component to local or binary component and that PVC to hold source code/binary needs to be created. Am I right @mik-dass?

Not a good name variable, but that is something that is probably outside of the scope of this PR.

Reply:
Yes it is used to indicate if a new supervisorD PVC should be created during the update or not e.g during the update from local/binary to git component. I have added a comment regarding this variable.